presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkInjectorFactory.java

```diff
@@ -46,29 +46,33 @@
     private final SparkProcessType sparkProcessType;
     private final Map<String, String> configProperties;
     private final Map<String, Map<String, String>> catalogProperties;
+    private final Optional<Map<String, String>> eventListenerProperties;
```

Comment:
I personally don't prefer to have `Optional<Collection<>>` as configuration, as the semantic difference between "not present" and "presented but empty" can be very tricky. Can we enforce `eventListenerProperties` always have to present? (`EVENT_LISTENER_PROPERTY_NAME` can be set to a special value such as "NO_EVENT_LISTENER")  ?

Reply:
I would rather prefer it to be explicitly. One can read it as event listener is not configured - `eventListenerProperties = Optional.empty()`. Or event listener is configured - the the `Optional` contains configuration properties for event listener.